### PR TITLE
Generate Scala Code from existing view and table defintions

### PR DESCRIPTION
**Pull request by hamnis**
First class codegen project with compile time verification that it compiles for every supported scala version

We can maybe use https://github.com/com-lihaoyi/PPrint as a codegen tool

### CODE REVIEW

File: core/src/main/scala/no/nrk/bigquery/BigQueryClient.scala
Comment (hamnis) [inline core/src/main/scala/no/nrk/bigquery/BigQueryClient.scala:1]: this should probably be its own PR

**Comment by hamnis** [comment]
Should resolve #183

File: zetasql/src/main/scala/no/nrk/bigquery/ZetaSql.scala
Comment (hamnis) [inline zetasql/src/main/scala/no/nrk/bigquery/ZetaSql.scala:1]: this should be its own PR

**Comment by oyvindberg** [comment]
This should work! 

We could do a lot with this, but the most important thing I had in mind was to use this to bootstrap an example project from public BQ data sets without too much typing 👍

File: build.sbt [hunk 1]
```diff
@@ -157,6 +156,34 @@ lazy val testing = crossProject(JVMPlatform)
     mimaBinaryIssueFilters := Nil
   )
 
+lazy val examples = crossProject(JVMPlatform)
```
[the remark this answers is not in the document]
Comment (oyvindberg) [reply build.sbt:159]: Codegen project, definitely. The codegen is the start of adoption likely, and it'll feel more serious if it's not some afterthought in the middle of an example folder. Also doesn't make sense to publish an example project

**Review by ingarabr — APPROVED** [review]
Looking good. Had been nice if you added some documentation.

**Review by HenningKoller — APPROVED** [review]
